frontend/src/Components/Player.jsx

```diff
@@ -18,9 +18,8 @@ export default function Player({ playerId, playerName, style, childRef }) {
     const playerStyle = classNames({
         [styles.playerWrapper]: true,
         [styles.isHoverable]: isHoverable,
-        [styles.hasVoted]: hasVoted,
+        [styles.hasVoted]: isDead ? isDead : hasVoted,
```

Comment:
This dosent seem to fix the issue for me? The hasVoted clasName is purely cosmetic (it puts a red dot next to the player) and dosent prevent a player from voting. The `isDead` boolean refers to the current player that you are clicking on (not you). To fix this, you have to check alivePlayers to see if your nickname is present. 

```
    const { state: generalState } = useContext(GeneralContext);
    const amIDead = gameState.alivePlayers.includes(generalState.nickname)
```
(This doesn't really look that clean so it may be better isDead as part of the game state)

Reply:
Agreed. We need a solution that prevents the dead user from being able to click other players. Otherwise the server will still receive vote information.

Reply:
@ateagit The solution does not work because "hasVoted" variable didn't do the job properly? Don't you think if one person is voted then he would not be able to vote again? So we should create another ticket to fix the hasVoted logic.

Reply:
This is up to the team. I'm ok with players being able to change their vote as long as it's before the night ends (with an exception being the detective). The `hasVoted` variable simply tells you if the current player object being rendered is in the `playersWhoVoted` list, it does not tell you if you (your player object) has voted. In other words, its only job is to add styling if that player has voted for another player during the vote phase. To check if you (the current player) has voted, you have to check if the `gameState.votingState.vote` variable is set (meaning that you have voted for someone). Always setting that for dead people dosent seem like a good workaround to prevent people from voting.

Reply:
> This is up to the team. I'm ok with players being able to change their vote as long as it's before the night ends (with an exception being the detective).

I agree with this behaviour! Unless we implement a "vote confirm" button, we should let players change their votes until the time runs out. Especially, after testing #198, I thought it was quite cool being able to click different players and seeing who I was voting for :)

Reply:
@ateagit @reeved So after implementing that, then we would put the dead users into the votedConfirmed group. From my perspective, every program should follow the DRY rule. If implement the 'cannot vote' logic twice, it will make the program WET.

Reply:
Yeah thats one solution. But the game has no "vote confirm" logic implemented in the front/backend, just a map for the current voting session that stores who has voted, and who they voted for. If we want to add vote confirm logic, then you would need to talk to the Backend team too because this would require additional work form both teams.

I think Alan's original suggestion was to use the ClickValidation that's already present in the Player Component. But do another check for `amIDead` which checks if the Player is alive or not. There is also an `isDead` but that is used purely for styling purposes for each Player rendered through the Table component.

This way, a dead player will not be able to click any other players AT ALL. So we wouldn't have to worry about them having the voted symbol.

Reply:
@reeved  so what about the "Another reason that I think we should change the votedConfirmed logic is that not only dead players cant vote but also a voted confirmed player. Correct? So the logic should be implemented for the votedConfirmed status instead of a dead player?" section. lol

Reply:
Preferably lets not create another vote confirm logic this late. Have we still not agreed on a solution to this issue? I thought its just check if player is dead and if so don't render the hover thing over other players and if in the case they do click don't send vote message across.